brave/src/main/java/brave/Tracer.java

```diff
@@ -495,6 +492,46 @@ public ScopedSpan startScopedSpan(String name) {
     return startScopedSpanWithParent(name, null);
   }
 
+  /**
+   * Like {@link #startScopedSpan(String)} except when there is no trace in process, the sampler
+   * {@link SamplerFunction#trySample(Object) triggers} against the supplied argument.
+   *
+   * @param name the {@link Span#name(String) span name}
+   * @param samplerFunction invoked if there's no {@link CurrentTraceContext#get() current trace}
+   * @param arg parameter to {@link SamplerFunction#trySample(Object)}
+   * @see #nextSpan(SamplerFunction, Object)
+   * @since 5.8
+   */
+  public <T> ScopedSpan startScopedSpan(String name, SamplerFunction<T> samplerFunction, T arg) {
+    if (name == null) throw new NullPointerException("name == null");
```

Comment:
This is the same as old code so probably ok, but just wondering should we also check if it's empty?

Reply:
I noticed some old examples actually pass empty.. we couldn't error, but we could eagerly coerce to null..